bokeh/charts/_builder.py

```diff
@@ -76,8 +81,26 @@ class Builder(HasProps):
 
     x_range = Instance(Range)
     y_range = Instance(Range)
+    y_names = Seq(String)
+    x_names = Seq(String)
 
     palette = Seq(Color, default=DEFAULT_PALETTE)
+    source = Instance(ColumnDataSource)
+
+    index = Any(help="""
+    An index to be used for all data series as follows:
+
+    - A 1d iterable of any sort that will be used as
+       series common index
+
+    - As a string that corresponds to the key of the
+       mapping to be used as index (and not as data
+       series) if area.values is a mapping (like a dict,
+       an OrderedDict or a pandas DataFrame)
+
+   """)
```

Comment:
Same comment basically. How do we fit in things like image charts(i.e. `pcolor`) if these assumptions are baked into every builder